### ISSUE TITLE
Allow combining stdout and stderr more easily

### ISSUE THREAD

**Issue by robx** [fpco/typed-process#48]
I'm not entirely sure what the best approach here would be, but right now it's easy to e.g. collect stdout and stderr individually as bytestrings via `byteStringOutput`, but I can't find a nice way to do the same with interleaved stdout&stderr.

What might help:
- export `byteStringFromHandle` to allow doing this manually; i.e. create pipe, pass the same handle as both stderr and stdout, and then the use can apply `byteStringFromHandle` to the write end themselves
- maybe there's a way to allow essentially `setStderr sameAsStdoutOutput`?

**Comment by robx** [comment]
I just found `readProcessInterleaved`, apologies. Somehow I thought `withProcess*` was the only thing I had to work with...

**Comment by tomjaguarpaw** [comment]
Perhaps the documentation needs improvement to draw attention away from `withProcess*`. If you can share how you came to the conclusion you did then that would be helpful (granted, that's not necessarily an easy thing to explain ...)

**Comment by robx** [comment]
My starting point: Working on a tool that already uses typed-process to spawn a subprocess (via `withProcessTerm`, redirecting output to a file). And I needed to do something similar but collecting the output in-process.

Then my path was:
- hackage docs page -> "contents" to get a general overview -> follow link to README -> see the `byteStringOutput` in the first worked example at https://github.com/fpco/typed-process#synopsis
- make that work with just stdout, figure out that the output I actually need is in stderr, and come up with a hacky solution that does a `stdout ++ stderr` and check that for substring (this is writing some tests, don't really care where it comes from)
- thinking this should be easier to do (bit of an XY problem, I was set on solving this using `byteStringOutput` and `withProcess`), I ended up filing the issue

Maybe something can be improved? I have to say though that `typed-process` is really well documented by comparison to most other.
